### PR TITLE
Parse a job as failure if there were top level excpetions, even if some tests were run

### PR DESCRIPTION
**Pull request by LesterTheTester**
We saw this recently in one of our runs:

screenshot from 2016-06-15 15 45 47

We have a top level exception present, but it occurred in the middle of running tests. So, as long as some tests have executed, the logic in parseOverallSuccess marks the run as a success. However, we haven't executed all the tests, so this should really be marked as a failure.  I think the logic here should be || instead of &&...can anyone justify why it's currently &&?

### CODE REVIEW

**Comment by edenman** [comment]
screen shot 2016-06-15 at 3 56 03 pm
Jake wrote that code three years ago, hahaha.  I guess I don't care?  @JakeWharton thoughts?

**Comment by JakeWharton** [comment]
:shipit:

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java [hunk 1]
```diff
@@ -269,8 +269,8 @@ static boolean parseOverallSuccess(SpoonSummary summary) {
       if (result.getInstallFailed()) {
         return false; // App and/or test installation failed.
       }
-      if (!result.getExceptions().isEmpty() && result.getTestResults().isEmpty()) {
-        return false; // No tests run and top-level exception present.
+      if (!result.getExceptions().isEmpty() || result.getTestResults().isEmpty()) {
+        return false; // Top-level exception present, or no tests were run
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java]: nit: missing period on comment.

[Comments should be sentences.][1]

 [1]: http://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

**Comment by LesterTheTester** [comment]
Ok, there's a test that asserts on the behaviour when no tests are run. I can either update that test so that it expects to fail when no tests are run, or remove the result.getTestResults().isEmpty() entirely from that comparison.  Preference?  Perhaps you have some use case where you want jobs with no tests to pass?

**Comment by edenman** [comment]
I think just remove it from the check entirely

**Comment by JakeWharton** [comment]
Isn't there an option to fail when empty?

**Comment by LesterTheTester** [comment]
Cool. Did that and added a test for summaries with both results and exceptions.

**Comment by edenman** [comment]
@JakeWharton you're maybe thinking of `--fail-if-no-device-connected`?  tangential concern

**Comment by JakeWharton** [comment]
An empty suite seems like you broke your runner and I wouldn't want to
allow that to pass.

**Comment by edenman** [comment]
screen shot 2016-06-15 at 4 38 44 pm
Past you disagrees, @JakeWharton.  I don't really care one way or the other, though...if you think we should change it, I'm fine w/that.

**Comment by JakeWharton** [comment]
Well look at me. I thought we changed it in the last year or so to do
that.. but perhaps I'm again thinking of the no device case.

I do feel like an empty suite is an error. Otherwise if you break your
runner your tests will always be green and you might not find out for a
while. It can be fixed separately (or just filed to be fixed).

**Comment by edenman** [comment]
Yeah, I'm sold.  @LesterTheTester up to you if you feel like making that change as part of this PR or not.

**Comment by LesterTheTester** [comment]
Yep...just updated. That was my inclination before but didn't want to step on tests :chicken:

**Comment by edenman** [comment]
1.5.6 released.  Thanks @LesterTheTester!

**Pull request merged**